### PR TITLE
Ns luis integration

### PR DESCRIPTION
**Pull request by DiegoAraque21**

### CODE REVIEW

File: src/listings/search/index.ts [hunk 1]
```diff
@@ -10,6 +15,12 @@ import { getGeolocation } from '../../helpers/geolocation'
 const INDEX = 'listings-dev-2'
 const MODEL_ID = 'sentence-transformers__msmarco-minilm-l-12-v3'
 const MAX_GEO_DISTANCE = '100km'
+const DEPLOYMENT_NAME = 'deploy_v1'
+const PROJECT_NAME = 'NER_NLSEARCH'
+
+interface Entities_mapped {
+  [key: string]: any
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:22]: why   `[key: string]: any` and not  ` [key: string]: string`

File: src/listings/search/index.ts [hunk 2]
```diff
@@ -92,8 +103,44 @@ const buildNeuralSearchQuery = (
   return searchQuery
 }
 
+const buildLuisEntitiesMap = (entities: Entity[]) => {
+  const entitiesMap: Entities_mapped = {}
+  for (const record of entities) {
+    if (record.confidenceScore >= 0.5) {
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:109]: can we create a constant for magic number 0.5, like `const NER_CONFIDENCE_THRESHOLD = 0.5`

File: src/listings/search/index.ts [hunk 3]
```diff
@@ -110,18 +157,55 @@ export const handler: APIGatewayProxyHandler = async (event): Promise<APIGateway
...
       rangeFilterClauses,
+      enableNer,
     } = getQueryParts(queryParameters.data)
+    let newFilters: TermQuery[] = []
+    // get entities
+    if (enableNer) {
+      if (!query) {
+        return badRequestResponse('Missing query parameter for ner search')
+      }
+      const actions: AnalyzeBatchAction[] = [
```
Comment (StephanGuingor) [inline src/listings/search/index.ts:168]: and maybe we could put al this ner logic in a function called, getCustomNEREntities(), wdyt?

Comment (DiegoAraque21) [reply src/listings/search/index.ts:168]: va perfecto. Creo una función en helpers

**Review by StephanGuingor — APPROVED** [review]

**Pull request merged**